apps/frontend/src/libs/modules/http/libs/exceptions/auth.exception.ts

```diff
@@ -0,0 +1,29 @@
+import { type APIErrorType } from "~/libs/enums/enums.js";
+import { type HTTPCode } from "~/libs/modules/http/http.js";
+import {
+	type APIValidationErrorDetail,
+	type ValueOf,
+} from "~/libs/types/types.js";
+
+import { HTTPError } from "./http-error.exception.js";
+
+type Constructor = {
+	cause?: unknown;
+	details: APIValidationErrorDetail[];
+	message: string;
+	name?: string;
+	status: ValueOf<typeof HTTPCode>;
+	type: ValueOf<typeof APIErrorType>;
+};
+
+class AuthError extends HTTPError {
+	public static readonly name = "Unauthorized";
+	public name: string;
```

Comment:
having it as a static and internal property is a bit confusing, let's only keep it as a property and use ExceptionName enum for matching. Also, instead of "Unauthorized", it's better to either name it 'AuthError' - to match the error, or to rename the error to Unauthorized error. I like the first option more
```suggestion
	public name: string;
```